### ISSUE TITLE
Scss / Sass compiles okay in dev but not generate – Nuxt 3

### ISSUE THREAD

**Issue by toddpadwick** [nuxt/framework#8665]

**Comment by manniL** [comment]
Hey! Could you kindly provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by toddpadwick** [comment]
Hey @manniL  I've just added a codesandbox link in description :)

**Comment by brokenfiles** [comment]
I have the exact same issue here. 
`
    css: [
        '@/assets/css/tailwind.css',
        '@/assets/css/layout.scss',
    ],
`
The tailwind.css file is loaded while the layout.scss (which is a scss file) is not loaded.
No errors when I run yarn generate and in the console. 

Everything works in dev env.

**Comment by zhoneyboo** [comment]
Any update to this one? I'm having issue with this one too.

**Comment by brokenfiles** [comment]
> Any update to this one? I'm having issue with this one too.

```js
    css: [
        '@fortawesome/fontawesome-svg-core/styles.css',
        '@/assets/css/tailwind.css',
        // '@/assets/css/layout.scss',
    ],

    vite: {
        css: {
            preprocessorOptions: {
                scss: {
                    additionalData: '@use "@/assets/css/layout.scss" as *;'
                }
            }
        }
    },
```

I removed the layout.scss from the CSS part and added a configuration to Vite.
But it's not ideal.

**Comment by toddpadwick** [comment]
@brokenfiles  that didn't work for me unfortunately, as I am using scss in all my components too, which requires globally accessible variables :(

**Comment by brokenfiles** [comment]
@toddpadwick 
In my `layouts/default.vue` i added an import, did you ?

```vue
<style scoped lang="scss">
@import 'assets/css/layout'; // <------ this line

#application {
  overflow: hidden;
}

</style>
```

**Comment by toddpadwick** [comment]
Oh okay. that seems pretty long-winded to have to do that in every component. With Nuxt 2, scss was globally accessible. I'm sure there must be something else at play here.

**Comment by brokenfiles** [comment]
I don't know why. It's the only solution I've found for the moment. 
Even if I want to use Nuxt3 in production, don't forget that it is still in release candidate.

**Comment by toddpadwick** [comment]
True, may have been risky of me to opt for Nuxt 3 for a production project, however, the official release is [scheduled for 16th November](https://v3.nuxtjs.org/community/roadmap/), which is pretty soon, so hopefully it'll be okay.

**Comment by brokenfiles** [comment]
@toddpadwick Yes I just saw the release date in the commits just now! I am very excited 🙌
Personally, I will wait for the official release to start my current project. I'll probably start working on the backend first :p

**Comment by denisasJersovas** [comment]
I have a same problem, found out solution at least for now
in your component you can import external style file like that and build works, let's wait until config compilation is fixed probably :) 
"<style scoped src="../assets/styles/scss/_navigation.scss" lang="scss"></style>"

**Comment by toddpadwick** [comment]
Thanks @denisasJersovas so sounds like this confirms that it is in fact a bug and not a config problem. 
I wonder if its related to [this other bug I am experiencing](https://github.com/nuxt/framework/issues/8911). Are you having this too?

**Comment by waysagency** [comment]
Hi @toddpadwick,

I sometimes have this problem... I created a static website with Nuxt 3 and sometimes when I navigate through the website it suddenly doesn't show anything except the footer and header.

I also have the .scss issue. The .css will be printed directly in a <style> tag within the header, but .scss doesn't work for now... Super annoying

**Comment by toddpadwick** [comment]
Thanks @waysagency at least i now know its not just bad code on my part. But its making me worried that I adopted nuxt 3 too early for a client project. I thought as it was so near to full release that it would be okay and really didn't want to be stuck with Vue 2 for any new projects.

**Comment by zhoneyboo** [comment]
Hi @toddpadwick - I just update my version to 3.0.0.rc13

**Comment by toddpadwick** [comment]
Updating to 3.0.0.rc13 didn't help for me. 
When I try and import the variables and scss utilities into the global scss as advised above, I get this error:
```
No response returned from render handler: /__nuxt_error?url=/blog/masterclasses/test-masterclass&statusCode=500&statusMessage=Vite+Error&message=[vite-node]+[plugin:vite:css]+[VITE_ERROR]+./assets/scss/tools.scss:1:1+%3Cbr%3E%3Cpre%3EError:+Module+loop:+this+module+is+already+being+loaded.%0A++%E2%95%B7%0A1+%E2%94%82+@use+%22@/assets/scss/tools.scss%22+as+*;%0A++%E2%94%82+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^%0A++%E2%95%B5%0A++assets/scss/tools.scss+1:1++root+stylesheet%3C/pre%3E%3Cbr%3E&stack=%3Cpre%3E%3Cspan+class=%22stack%22%3Eat+./assets/scss/tools.scss:1:1%3C/span%3E%3C/pre%3E&data
```
(this module is already being added).

So this suggests, in dev mode, vite is already doing what it should be doing and importing the scss as normal. It just doesn't in build :(

**Comment by toddpadwick** [comment]
Hey @danielroe , as I added a reproduction to the description, could you update the label?

**Comment by toddpadwick** [comment]
Turns out @zhoneyboo was correct – updating to RC13 fixes this – however, I had not updated all other dependencies so might not have initially fixed it. had to upgrade all my packages for it to work.

@brokenfiles although your suggestion worked in the interim, I would recommend removing it as with me, I discovered it meant my CSS was being compiled twice, doubling the size of the payload. 

the following should work now:
```
css:[
		"@/assets/scss/main.scss" // contains all global styles
	],
	vite: {
        css: {
            preprocessorOptions: {
                scss: {
                    additionalData: '@use "@/assets/scss/tools.scss" as *;' // contains just variables/mixins/functions needed for components and global styles ( do not include any compilable css)
                },
            },
        },
    },
```

@danielroe  this can be closed. thank you!
